src/adapter/src/coord/statement_logging.rs

```diff
@@ -70,6 +77,10 @@ pub(crate) struct StatementLogging {
     /// Only used by tests; otherwise, `rand::thread_rng()` is used.
     /// Controlled by the system var `statement_logging_use_reproducible_rng`.
     reproducible_rng: rand_chacha::ChaCha8Rng,
+
+    pending_statement_execution_events: Vec<(Row, Diff)>,
+    pending_prepared_statement_events: Vec<Row>,
+    pending_session_events: Vec<Row>,
```

Comment:
just a thought: if these can be collections end up being very large, it might be nice to use something like [SegVec](https://docs.rs/segvec/latest/segvec/)